nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/AbstractGCPubSubProcessor.java

```diff
@@ -36,16 +37,47 @@
 
 public abstract class AbstractGCPubSubProcessor extends AbstractGCPProcessor implements VerifiableProcessor {
 
-    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+    public static final PropertyDescriptor BATCH_SIZE_THRESHOLD = new PropertyDescriptor.Builder()
             .name("gcp-pubsub-publish-batch-size")
-            .displayName("Batch Size")
+            .displayName("Batch Size Threshold")
             .description("Indicates the number of messages the cloud service should bundle together in a batch. If not set and left empty, only one message " +
                     "will be used in a batch")
             .required(true)
             .defaultValue("15")
             .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
             .build();
 
+    public static final PropertyDescriptor BATCH_BYTES_THRESHOLD = new PropertyDescriptor.Builder()
+            .name("gcp-batch-bytes")
+            .displayName("Batch Bytes Threshold")
+            .description("Publish request gets triggered based on this Batch Bytes Threshold property and"
+                    + " the " + BATCH_SIZE_THRESHOLD.getDisplayName() + " property, whichever condition is met first.")
+            .required(true)
+            .defaultValue("3 MB")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor BATCH_DELAY_THRESHOLD = new PropertyDescriptor.Builder()
+            .name("gcp-pubsub-publish-batch-delay")
+            .displayName("Batch Delay Threshold")
+            .description("Indicates the delay threshold to use for batching. After this amount of time has elapsed " +
+                    "(counting from the first element added), the elements will be wrapped up in a batch and sent. " +
+                    "This value should not be set too high, usually on the order of milliseconds. Otherwise, calls " +
+                    "might appear to never complete.")
+            .defaultValue("100 millis")
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PUBSUB_ENDPOINT = new PropertyDescriptor
+            .Builder().name("pubsub-api-endpoint")
+            .displayName("PubSub API Endpoint")
+            .description("Override the gRPC endpoint (form is \"host:port\").")

```

Comment:
Recommend adjusting the example to avoid escaping.
```suggestion
            .description("Override the gRPC endpoint in the form of [host:port]")

```